docs/lang/articles/basic/field.md

```diff
@@ -3,103 +3,128 @@ sidebar_position: 4
 ---
 
 # Fields
-Taichi fields are used to store data.
-In general, fields are global data containers that can be read and written from both the Python scope and the Taichi scope.
 
-A field has its own data type and shape and can be considered as a multi-dimensional array of elements.
-An element of a field can be a **scalar**, a **vector**, a **matrix**, or a **struct**.
-The sparsity of a field element is **dense** by default, but it can also be **sparse**, as detailed described in [Sparse spatial data structures](/lang/articles/advanced/sparse.md).
+The term _field_ is borrowed from mathematics and physics. If you already know [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (for example heat field), or vector field (for example [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)), then it is straightforward for you to understand fields in Taichi.
 
-:::note
-The term **field** is borrowed from mathematics and physics.
-If you have already known [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (e.g., heat field) or vector field (e.g., [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)) in mathematics and physics,
-it will be straightforward to understand the fields in Taichi.
-:::
+Fields in Taichi are the global data containers that can be accessed from both the Python scope and the Taichi scope. Just like an ndarray in NumPy or a tensor in PyTorch, a field in Taichi is defined as a multi-dimensional array of elements. Each of its elements can be a scalar, vector, matrix, struct, or pointer.
+
+- You can use a field as a sparse data structure, with each of its elememt being a pointer pointing to a different field either sparse or dense. And this pattern can go on. See the [Sparse spatial data structures](/lang/articles/advanced/sparse.md) for more information.
```

Comment:
The new description is not accurate: the element of a field can only be a scalar, a vector/matrix or a struct. 

> An element of a field can be a **scalar**, a **vector**, a **matrix**, or a **struct**.

Pointer only exists at the SNode hierarchy level.

Reply:
This is my fault, I though in taichi we can use fields to store points to other dense structures.

Reply:
Yup, that is the **implementation** of field. Field itself is just an opaque multi-dimensional array.